### PR TITLE
#97, #470 : add UI elements for queue report downloads

### LINKED ISSUE #97: Self-service reporting by queue
In order to meet the potential use case of lobby management, as well as to respond to general inevitable requests for reporting capabilities, Remote Office Hours Queue should make it possible for users to pull reports about usage of their queue. Some elements that may be important to report on include:

- Most popular times of the day, days of the week, and weeks of the year
- Number of attendees seen by each host
- Most popular words used in the agenda field
- Number of unique attendees in a period of time

Reporting could be implemented by sending Caliper events to the Unizin Data Platform (UDP) and providing reports that utilize the Google Cloud Platform connection to the UDP.

### PR DESCRIPTION
**Pull request by jaydonkrooss**
UI elements required for #97 ( as well as #470 ) including "download log" buttons for individual queues and download for all queues. Both can be found on the Manage Queues page.

### CODE REVIEW

**Comment by jxiao21** [comment]
Buttons look good to me. Minor style change suggestion for aesthetic purposes: is it possible to vertically center the badges (queue id, status) and queue name in their respective boxes? Right now on my end each of those three things are a little above the center of their boxes. (This isn't a change that needs to be made in this PR, but just a suggestion.)
Screenshot 2024-06-14 at 12 18 59 PM

**Comment by jaydonkrooss** [comment]
@jxiao21 good suggestion to clean up the spacing I introduced here, I went ahead and added some CSS for center aligning the rows.

**Comment by zqian** [comment]
@jaydonkrooss Could you please sync up the PR with the latest code? So that we can test the full download process?

**Comment by zqian** [comment]
Number '0' was shown next to the "+ Add Queue" button with zero queue item.

image

File: src/assets/src/components/manage.tsx [hunk 1]
```diff
@@ -26,6 +30,9 @@ function ManageQueueTable(props: ManageQueueTableProps) {
                 <Link to="/add_queue" tabIndex={-1}>
                     <Button variant='success' aria-label='Add Queue'>+ Add Queue</Button>
                 </Link>
+                {props.queues.length && props.onAllQueueHistoryDownload &&(
```
Comment (zqian) [inline src/assets/src/components/manage.tsx:33]: Number '0' was shown next to the "+ Add Queue" button with zero queue item.

image

**Review by zqian — APPROVED** [review]
The download links work as expected. I left a comment for an edge case.

File: src/assets/src/components/manage.tsx (hunk 1 above)
Comment (jaydonkrooss) [reply src/assets/src/components/manage.tsx]: I noticed this too. Addressed in the recent commit, the button should be available to press even if there's no queues in the UI

**Pull request merged**